inference-engine/src/inference_engine/threading/ie_cpu_streams_executor.cpp

```diff
@@ -66,37 +66,74 @@ struct CPUStreamsExecutor::Impl {
-                    ((_impl->_config._streams + _impl->_usedNumaNodes.size() - 1)/_impl->_usedNumaNodes.size()))
-                : _impl->_usedNumaNodes.at(_streamId % _impl->_usedNumaNodes.size());
 #if IE_THREAD == IE_THREAD_TBB || IE_THREAD == IE_THREAD_TBB_AUTO
-            auto concurrency = (0 == _impl->_config._threadsPerStream) ? tbb::task_arena::automatic : _impl->_config._threadsPerStream;
-            if (ThreadBindingType::NUMA == _impl->_config._threadBindingType) {
-#if TBB_INTERFACE_VERSION >= 11100  // TBB has numa aware task_arena api
-                _taskArena.reset(new tbb::task_arena{tbb::task_arena::constraints{_numaNodeId, concurrency}});
-#else
-                _taskArena.reset(new tbb::task_arena{concurrency});
-#endif
-            } else if ((0 != _impl->_config._threadsPerStream) || (ThreadBindingType::CORES == _impl->_config._threadBindingType)) {
-                _taskArena.reset(new tbb::task_arena{concurrency});
-                if (ThreadBindingType::CORES == _impl->_config._threadBindingType) {
-                    CpuSet processMask;
-                    int    ncpus = 0;
-                    std::tie(processMask, ncpus) = GetProcessMask();
-                    if (nullptr != processMask) {
-                        _observer.reset(new Observer{*_taskArena,
-                                                     std::move(processMask),
-                                                     ncpus,
-                                                     _streamId,
-                                                     _impl->_config._threadsPerStream,
-                                                     _impl->_config._threadBindingStep,
-                                                     _impl->_config._threadBindingOffset});
-                        _observer->observe(true);
+            #if TBB_INTERFACE_VERSION >= 12010 // TBB with hybrid CPU aware task_arena api
+            auto core_types = oneapi::tbb::info::core_types();
+            if (core_types.size() > 1 /*Hybrid CPU*/
+                 && oneapi::tbb::info::efficiency(*core_types.begin()) != -1 /* hwloc recognized relative cores efficiency */) {
+                if (_impl->_config._streams == 1) {
+                    auto big_cores = core_types.back(); // latency default is runing on Big cores only
+                    auto concurrency = _impl->_config._threadsPerStream;
+                    printf("%s, LATENCY CASE, StreamId: %d (%d threads) assigned CORE TYPE : %d (CONCURRENCY: %d) \n",
+                        _impl->_config._name.c_str(), _streamId, _impl->_config._threadsPerStream,
+                        static_cast<int>(big_cores), concurrency);
+                    _taskArena.reset(new tbb::task_arena{tbb::task_arena::constraints{big_cores, concurrency}});
+                } else {
+                    // TODO: wrap #streams/threads over core types
+                    int threads_needed = (_streamId+1)*_impl->_config._threadsPerStream;
```

Comment:
these are generaly valid points (on keeping the affinity logic in the config population).
I will need to brush the code anyway.
On the exposing explicit Big/Little cores placement API. I don't think this is needed now.

Finally, on the point that devices' plugins may want to use Little cores (only) - I don't think this is valid.
Instead, the devices' CPUExecutors (other than actual CPU inference one) should NOT affinitize themselves to the specific type of cores (to let the OS or HGS+ move thier threads to the Big Cores, whenever the latter are vacant).
So by default the "cores-types" config param (that we are discussing here) should not be set to any specific type. Do you agree?.

Reply:
I moved the init phase to the costructor of the CPUStreamExecutor::Impl, so that on stream's creation only a single lookup to the map (kind of upper_bound logic) is made (unless the big cores are forced to use, which is even shorter code_